lib/miq_pglogical/connection_handling.rb

```diff
@@ -2,10 +2,37 @@ class MiqPglogical
   module ConnectionHandling
     extend ActiveSupport::Concern
 
+    included do
+      delegate :logical_replication_supported?, :to => :class
+    end
+
     module ClassMethods
+      def logical_replication_supported?
+        return @logical_replication_supported if defined?(@logical_replication_supported)
+
+        is_superuser = ActiveRecord::Base.connection.select_value("SELECT usesuper FROM pg_user WHERE usename = CURRENT_USER;", "SQL")
+        unless is_superuser
+          warn_bookends = ["\e[33m", "\e[0m"]
+          msg = "WARNING: Current user is NOT a superuser, logical replication will not function."
+          if $stderr.tty?
+            warn(warn_bookends.join(msg).to_s)
```

Comment:
What a clever way to do this!

Reply:
I tried other ways in irb and this just came to me... it's so much better than the interpolation I had originally.

Reply:
I think the .to_s is redundant though, so if you're doing a cleanup in a follow up you can drop that.